README.md

```diff
@@ -144,20 +144,36 @@ number.
 
 ### Caching dependencies downloaded by Deno automatically
 
-Dependencies installed by Deno can be cached automatically, which is similar to
-the [`cache` option in `setup-node`](https://github.com/actions/setup-node).
+Dependencies installed by Deno can be cached automatically.
 
-To enable the cache, use `cache: true`. It's recommended to also add the
-`cache-hash` property, to scope caches based on lockfile changes.
+To enable the cache, use `cache: true`.
 
 ```yaml
 - uses: denoland/setup-deno@v2
   with:
     cache: true
-    cache-hash: ${{ hashFiles('**/deno.lock') }}
 ```
 
 > [!WARNING]
 > If an environment variable `DENO_DIR` is set for steps that run/download
 > dependencies, then `DENO_DIR` must also be set for the `denoland/setup-deno`
 > action, for the caching to work as intended.
+
+By default, the cache is automatically keyed by:
+
+- the github
+  [job_id](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_id)
+- the runner os and architecture
+- a hash of the deno.lock files in the project
+
+It is possible to customize the default hash (`${{ hashFiles('/deno.lock') }}`)
+used as part of the cache key via the `cache-hash` input.
+
+```yaml
+- uses: denoland/setup-deno@v2
+  with:
+    # setting `cache-hash` implies `cache: true`
+    # and will replace the default cache-hash
+    # of `${{ hashFiles('/deno.lock') }}`
```

Comment:
Minor nit: Perhaps add `**/` to show the exact glob used by default, and that it's not only the workspace root `/deno.lock` file.

```suggestion
It is possible to customize the default hash (`${{ hashFiles('**/deno.lock') }}`)
used as part of the cache key via the `cache-hash` input.

```yaml
- uses: denoland/setup-deno@v2
  with:
    # setting `cache-hash` implies `cache: true`
    # and will replace the default cache-hash
    # of `${{ hashFiles('**/deno.lock') }}`
```

Reply:
Thanks! yeah, it should be this.